core/src/loader.rs

```diff
@@ -1696,6 +1696,10 @@ impl<'gc> Loader<'gc> {
 
                     if let Some(domain) = domain {
                         library.set_avm2_domain(domain);
+                    } else {
+                        // This is in case the MovieLoaderData is AVM1, but loaded an AVM2 SWF (mixed AVM).
+                        let stage_domain = activation.context.avm2.stage_domain();
```

Comment:
Can you move this to the `let domain = if let MovieLoaderVMData::Avm2 {` block above (the `else` case)?